### ISSUE TITLE
Scrollbar issue

### ISSUE THREAD

**Issue by jwintz** [sebastiencs/company-box#8]
Hi, huge fan of `company-box` and definitely wanna try `sidebar.el`.

However, I can't get `icons-in-terminal` working on High Sierra (maybe my fault, I hacked the CoreText stuff to get `all-the-icons` working).

Anyway, using:

```lisp
(when (display-graphic-p)
  (def-package! company-box
    :hook (company-mode . company-box-mode)
    :init
    (setq company-box-icons-elisp
          (list
           (concat (all-the-icons-material "functions") " ")
           (concat (all-the-icons-material "check_circle") " ")
           (concat (all-the-icons-material "stars") " ")
           (concat (all-the-icons-material "format_paint") " ")))
    (setq company-box-icons-unknown (concat (all-the-icons-material "find_in_page") " "))
    (setq company-box-backends-colors nil)))
```

I've got a scroller issue, that is, all candidates cannot be visited.

Also, whenever the scrollbar is displayed, left hand side icons are cropped.

<img width="765" alt="screen shot 2018-04-11 at 23 27 53" src="https://user-images.githubusercontent.com/3277054/38644389-9fecfe44-3de0-11e8-8629-ad8daf97cba7.png">

**Comment by sebastiencs** [comment]
Thanks :)
You may also like https://github.com/sebastiencs/omnibox

> However, I can't get icons-in-terminal working on High Sierra (maybe my fault, I hacked the CoreText stuff to get all-the-icons working).

[This comment](https://github.com/sebastiencs/icons-in-terminal/issues/1#issuecomment-318905760) and the next one could probably help you to install it on mac os. It explains how to use `icons-in-terminal` in terminals, but if you just want to use the font on (emacs) GUI, you just need to install the font file like you would install any other font on your system (the `ttf` file in the build/ directory) 

> I've got a scroller issue, that is, all candidates cannot be visited.

What do you mean exactly ? You can't access some candidates ?

> Also, whenever the scrollbar is displayed, left hand side icons are cropped.

Can you provide an image ?

**Comment by jwintz** [comment]
Ok, thanks for the comment, I have installed fonts *I did not already have using __all-the-icons__*, e.g. `build/icons-in-terminal`, and some fonts from `icons-in-terminal` fonts folder, and hacked `/System/Library/Frameworks/ApplicationServices.framework/Frameworks/CoreText.framework/Resources/DefaultFontFallbacks.plist` to be aware of them.

So far `print-icons.sh` fails on Mac:

<img width="1032" alt="screen shot 2018-04-12 at 21 40 21" src="https://user-images.githubusercontent.com/3277054/38700049-1ec19a18-3e9a-11e8-9764-8550c92f6f33.png">

But the python script from the thread works fine:

<img width="1032" alt="screen shot 2018-04-12 at 21 41 00" src="https://user-images.githubusercontent.com/3277054/38700079-3621028e-3e9a-11e8-83ee-ca87b2c0f650.png">

It should however work in both situations, and we, MacOS Users would very much like to avoid hacking a file that requires to disable System Integrity Protection.

However, back to `company-box` :-)

Starting with `emacs-lisp-mode`:

<img width="747" alt="screen shot 2018-04-12 at 21 45 16" src="https://user-images.githubusercontent.com/3277054/38700290-d97b71e4-3e9a-11e8-935e-0e492864ecfe.png">

See the size of the scroll bar, and consider that I'm a the bottom of provided candidates.

Now for `c++-mode`:

<img width="692" alt="screen shot 2018-04-12 at 21 48 42" src="https://user-images.githubusercontent.com/3277054/38700432-4e0b9f7a-3e9b-11e8-9ca3-55810bd91345.png">

I only have few candidates as opposed to thousands when I do not use `company-box`.

**Comment by sebastiencs** [comment]
On the first image you can see that there are 2 child frames for company (3 with the doc). Looks like you have both `company-childframe` and `company-box` enabled at the same time ? 
Is one front-end not enough ? :)
You need to disable `company-childframe`.

Does it somehow solves the issue for the 2nd image too ?

**Comment by nealot** [comment]
I had a similar issue with Doom-Emacs and removing the default front-ends solved the overlap issue. However, using the tab to select elements still results in the scroll bar not following the selection. The scrollbar works fine with C-n and C-p, though.

**Comment by sebastiencs** [comment]
@nealot What the tab key, ivy-scroll-up and ivy-scroll-down are supposed to do, exactly ? Do theses commands work with the built-in front-end ?

**Comment by nealot** [comment]
@sebastiencs Yeah, Tab is supposed to work as C-n essentially and ivy-scroll-up and ivy-scroll-down scroll the menu up or down by a large value. They actually still work but the scroll bar doesn't display their results if they leave the scroll area. 

Btw, do you know if that margin at the bottom is intentional? Anyway I could get rid of it? It looks out of place.

**Comment by nealot** [comment]
@jwintz With company box, the number of elements available is controlled by company-tooltip-limit so you'll have to add something like `setq company-tooltip-limit 1000` to see 1000 results. That looks something like this:
![screenshot-2018-04-16_17-12-59](https://user-images.githubusercontent.com/36805289/38835537-7c345e72-4199-11e8-9517-a98fa3895414.png)
It would be better to have both a limit for the number of elements displayed and for the number of elements able to be selected via scrolling.

**Comment by dgutov** [comment]
> you'll have to add something like setq company-tooltip-limit 1000 to see 1000 results

That's an odd use of this variable...

**Comment by sebastiencs** [comment]
@nealot I will rework the keymap of company-box and add commands to scroll by large value.

I see in your pictures that the start of candidates (after the icons) isn't aligned with the completion prefix in your current buffer (see the readme picture).
Can you disable `display-line-numbers-mode` and see if it still occurs ?

There is a bug that is fixed in emacs master branch but I thought it happened only when the completion prefix is on the very beginning of line: http://lists.gnu.org/archive/html/bug-gnu-emacs/2018-03/msg00509.html

>> you'll have to add something like setq company-tooltip-limit 1000 to see 1000 results

> That's an odd use of this variable...

Yes I misunderstood the variable, I will change that

**Comment by nealot** [comment]
The issue has been resolved for me with the recent commits.
